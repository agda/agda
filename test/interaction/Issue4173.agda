-- Andreas, 2019-11-06, issue #4173, reported and testcase by nad.
-- Allow pattern matching on erased arguments in erased context.

-- {-# OPTIONS -v tc.cover.split:60 #-}

open import Agda.Builtin.Bool

@0 F : @0 Bool → Set₁
F true  = Set
F false = Set

-- Should succeed.

@0 G : @0 Bool → Set₁
G x = {! x !}  -- splitting on x should also succeed
