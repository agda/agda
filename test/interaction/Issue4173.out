(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Set‚ÇÅ " nil)
((last . 1) . (agda2-goals-action '(0)))
((last . 2) . (agda2-make-case-action '("G false = ?" "G true = ?")))
((last . 1) . (agda2-goals-action '(0)))
